history/tools.py

```diff
@@ -83,39 +89,48 @@ def get_exchange_rate_btc_to_usd(d=None):
         d = datetime.datetime.now()
 
     exchange_pair = 'USDT_BTC'
-    latest_price = Price.objects.filter(symbol=exchange_pair,created_on__lt=d).order_by('-created_on').first()
+    latest_price = Price.objects.filter(symbol=exchange_pair, created_on__lt=d).order_by('-created_on').first()
     return latest_price.price
 
-def get_cost_basis(total_amount,symbol):
+
+def get_cost_basis(total_amount, symbol):
     from history.models import Trade
     running_amount = total_amount
     cost_basis_array = []
-    ts = Trade.objects.filter(symbol=symbol).order_by('-created_on').values_list('amount','price')
+    ts = Trade.objects.filter(symbol=symbol).order_by('-created_on').values_list('amount', 'price')
     for t in ts:
         if running_amount < 0:
             continue
```

Comment:
Maybe it's better to not enter the for loop when running_amount < 0 to avoid unnecessary looping. Something like this can be done:
```python
if running_amount > 0:
    for t in ts:
        ....
```